test/controllers/api/v2/smart_proxies_controller_test.rb

```diff
@@ -6,7 +6,7 @@ class Api::V2::SmartProxiesControllerTest < ActionController::TestCase
   include SmartProxiesControllerSharedTest
 
   setup do
-    stub_smart_proxy_v2_features
+    stub_smart_proxy_v2_features_and_statuses
```

Comment:
This masks the network requests that it performs. I think we should be much more specific and only stub those out where we expect them.

Reply:
Just to make sure I followed, do you think I need to keep lines 173,174 only? 
https://github.com/theforeman/foreman/blob/62228620c4265f98bfe5f8eb479a915d87fc94df/test/active_support_test_case_helper.rb#L174

Reply:
As explained in https://github.com/theforeman/foreman/pull/8572#discussion_r670387220: this is hiding a network access call. IMHO it should not be allowed to do this by default.

I do think this should also be hidden behind an `include_version` parameter. Then I'd like to see a specific test that does this access. Only in that test should it stub.

Reply:
Please revert this line to what it was and stub only in the test where it should be allowed.